collector/aws/elasticsearch.go

```diff
@@ -0,0 +1,304 @@
+package aws
+
+import (
+	"finala/collector"
+	"finala/collector/config"
+	"finala/expression"
+	"finala/interpolation"
+	"fmt"
+	"time"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	elasticsearch "github.com/aws/aws-sdk-go/service/elasticsearchservice"
+	"github.com/aws/aws-sdk-go/service/pricing"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// describeElasticsearchDomainsDefaultLimit is the number AWS limits to describe number of ES Cluster
+	describeElasticsearchDomainsDefaultLimit = 5
+)
+
+// ElasticSearchClientDescriptor defines the ElasticSearch client
+type ElasticSearchClientDescriptor interface {
+	DescribeElasticsearchDomains(*elasticsearch.DescribeElasticsearchDomainsInput) (*elasticsearch.DescribeElasticsearchDomainsOutput, error)
+	ListDomainNames(*elasticsearch.ListDomainNamesInput) (*elasticsearch.ListDomainNamesOutput, error)
+	ListTags(*elasticsearch.ListTagsInput) (*elasticsearch.ListTagsOutput, error)
+}
+
+// ElasticSearchManager will hold the ElasticSearch Manger strcut
+type ElasticSearchManager struct {
+	collector          collector.CollectorDescriber
+	client             ElasticSearchClientDescriptor
+	cloudWatchCLient   *CloudwatchManager
+	pricingClient      *PricingManager
+	metrics            []config.MetricConfig
+	accountID          string
+	region             string
+	namespace          string
+	servicePricingCode string
+	Name               string
+}
+
+// DetectedElasticSearch defines the detected AWS Elasticsearch cluster
+type DetectedElasticSearch struct {
+	Metric        string
+	Region        string
+	InstanceType  string
+	InstanceCount int64
+	collector.PriceDetectedFields
+}
+
+// elasticSearchVolumeType will hold the available volume types for ESCluster EBS
+var elasticSearchVolumeType = map[string]string{
+	"gp2":      "GP2",
+	"standard": "Magnetic",
+	"io1":      "PIOPS",
+}
+
+// NewElasticSearchManager implements AWS GO SDK
+func NewElasticSearchManager(collector collector.CollectorDescriber, client ElasticSearchClientDescriptor, cloudWatchCLient *CloudwatchManager, pricing *PricingManager, metrics []config.MetricConfig, region string, accountID string) *ElasticSearchManager {
```

Comment:
you can remove the `string` after the region and keep the `string` after the `accoundID`

Reply:
removed.